src/app/cli/src/init/graphql_internal.ml

```diff
@@ -235,51 +235,106 @@ struct
-                     only caller is `create_graphql_server`, which enforces that invariant
+        let invalid_uuid () =
+          respond_string ~status:`Precondition_failed
+            ~body:"Invalid server Uuid" ()
+        in
+        let invalid_sequence_no () =
+          respond_string ~status:`Precondition_failed
+            ~body:"Invalid sequence number" ()
+        in
+        let missing_sequence_info () =
+          respond_string ~status:`Precondition_failed
+            ~body:"Missing sequence information" ()
+        in
+        match auth_keys with
+        | None ->
+            respond ()
+        | Some pks -> (
+            (* auth_keys is Some iff authentication is required
+               only caller is `create_graphql_server`, which enforces that invariant
+            *)
+            match Cohttp.Header.get headers "Authorization" with
+            | None ->
+                unauthorized ()
+            | Some auth -> (
+                let open Core in
+                let verify_signature_and_respond pk signature =
+                  (* we need to know not only that the pk verifies the
+                     signature, but also that this pk is allowed to verify
+                     the signature; otherwise, the client could sign a
+                     request using an arbitrary pk
                   *)
-                  match Cohttp.Header.get headers "Authorization" with
-                  | None ->
-                      unauthorized ()
-                  | Some auth -> (
-                      let open Core in
-                      match String.split_on_chars auth ~on:[ ' ' ] with
-                      | [ "Signature"; pk_str; signature ] -> (
-                          match Itn_crypto.pubkey_of_base64 pk_str with
-                          | Ok pk ->
-                              (* we need to know not only that the pk verifies the
-                                 signature, but also that this pk is allowed to verify
-                                 the signature; otherwise, the client could sign a
-                                 request using an arbitrary pk
-                              *)
-                              if
-                                List.mem pks pk ~equal:Itn_crypto.equal_pubkeys
-                                && Itn_crypto.verify ~key:pk ~msg:body_str
-                                     signature
-                              then respond ()
-                              else unauthorized ()
-                          | Error _err ->
-                              (* not a base64-encoded key *)
-                              unauthorized () )
-                      | _ ->
-                          (* invalid auth scheme *)
-                          unauthorized () ) ) ) )
+                  if
+                    List.mem pks pk ~equal:Itn_crypto.equal_pubkeys
+                    && Itn_crypto.verify ~key:pk ~msg:body_str signature
```

Comment:
Crucially for replay protection, we need `msg` to be a concatenation of `uuid`, `seqno` and `body`.
Given that `uuid` and `seqno` are serialized to fixed number of bytes (16 for `uuid` and 4 for `seqno`) plain byte concatentaion should do just fine.